### ISSUE TITLE
Is it possible to create MetricServer using only a port input in non elevated permissions? 

### ISSUE THREAD

**Issue by Tal-L** [prometheus-net/prometheus-net#158]
First, thanks for a ver useful project! 

It seems that creating a MetricServer with port only (i.e. where hostname resolves to "+") is possible only in elevated permissions.

Is there another way to get it running?

This is the ctor I was using:
public MetricServer(int port, string url = "metrics/", CollectorRegistry registry = null, bool useHttps = false) : this("+", port, url, registry, useHttps) { }

Thanks a lot

**Comment by sandersaares** [comment]
Yes, it is possible to grant non-elevated users the right to open this port. The readme contains the instructions: https://github.com/prometheus-net/prometheus-net#publishing-via-standalone-http-handler

In short, the command is:

```
netsh http add urlacl url=http://+:1234/metrics user=DOMAIN\user
```

Alternatively, you may be able to avoid this by using a different metric server implementation (e.g. Kestrel or IIS hosted). I am not 100% up to date on which implementations require which permissions/elevation.

**Comment by Tal-L** [comment]
Thanks a lot for the help and for the quick reply.
Tal

**Comment by sandersaares** [comment]
I will mark this closed for now. If you have any more questions or if this topic does not appear resolved, feel free to leave a new comment or open a new issue.

**Comment by Tal-L** [comment]
Kestrel option works as suggested, which is great, thanks!

Seems like Prometheus.NetStandard.dll is missing in prometheus-net.AspNetCore. 
Building it locally and adding a reference to it works. 
Is it a known issue?

Thanks a lot,
Tal

**Comment by sandersaares** [comment]
> Seems like Prometheus.NetStandard.dll is missing in prometheus-net.AspNetCore.

The readme indicates that prometheus-net is to be installed via NuGet. Are you installing the app via the NuGet commands indicated? Other ways to install may require manual actions to install.

**Comment by Tal-L** [comment]
Hi

Yes, via NuGet:
![image](https://user-images.githubusercontent.com/6724720/66025533-62639e80-e4ff-11e9-97d6-50a64ddeb705.png)

Thanks

**Comment by sandersaares** [comment]
Thank you for the screenshot! That is very helpful. Yes, you have discovered some packaging defect. I will investigate.

**Comment by sandersaares** [comment]
My bad - insufficient testing before release! Forgot a line in the nuspec file. Fixed version 3.2.1 is now uploading to NuGet!

**Comment by Tal-L** [comment]
Great, works now. Thanks for the quick fix!
